Signal/src/util/OWSBackup.m

```diff
@@ -3,31 +3,38 @@
 //
 
 #import "OWSBackup.h"
-#import "NSNotificationCenter+OWS.h"
-#import "OWSBackupExport.h"
-#import <SignalServiceKit/AppContext.h>
-#import <SignalServiceKit/NSDate+OWS.h>
-#import <SignalServiceKit/OWSBackupStorage.h>
-#import <SignalServiceKit/OWSFileSystem.h>
-#import <SignalServiceKit/TSAccountManager.h>
+#import "OWSBackupExportJob.h"
+#import "OWSBackupImportJob.h"
+#import "Signal-Swift.h"
+#import <Curve25519Kit/Randomness.h>
 #import <SignalServiceKit/YapDatabaseConnection+OWS.h>
 
 NSString *const NSNotificationNameBackupStateDidChange = @"NSNotificationNameBackupStateDidChange";
 
 NSString *const OWSPrimaryStorage_OWSBackupCollection = @"OWSPrimaryStorage_OWSBackupCollection";
 NSString *const OWSBackup_IsBackupEnabledKey = @"OWSBackup_IsBackupEnabledKey";
+NSString *const OWSBackup_BackupKeyKey = @"OWSBackup_BackupKeyKey";
 NSString *const OWSBackup_LastExportSuccessDateKey = @"OWSBackup_LastExportSuccessDateKey";
 NSString *const OWSBackup_LastExportFailureDateKey = @"OWSBackup_LastExportFailureDateKey";
 
 NS_ASSUME_NONNULL_BEGIN
 
 // TODO: Observe Reachability.
-@interface OWSBackup () <OWSBackupExportDelegate>
+@interface OWSBackup () <OWSBackupJobDelegate>
 
 @property (nonatomic, readonly) YapDatabaseConnection *dbConnection;
 
 // This property should only be accessed on the main thread.
-@property (nonatomic, nullable) OWSBackupExport *backupExport;
+@property (nonatomic, nullable) OWSBackupExportJob *backupExportJob;
+
+// This property should only be accessed on the main thread.
+@property (nonatomic, nullable) OWSBackupImportJob *backupImportJob;
+
+@property (nonatomic, nullable) NSString *backupExportDescription;
+@property (nonatomic, nullable) NSNumber *backupExportProgress;
+
+@property (nonatomic, nullable) NSString *backupImportDescription;
+@property (nonatomic, nullable) NSNumber *backupImportProgress;
```

Comment:
We'll eventually surface this state in the UI so users can observe import/export "state" and "progress".